src/parser.rs

```diff
@@ -1109,35 +1197,57 @@ impl<'a> Parser<'a> {
                     }
                 }
             }
-            match &self.serialization[segment_start..] {
+
+            let segment_before_slash = if ends_with_slash {
+                &self.serialization[segment_start..self.serialization.len() - 1]
+            } else {
+                &self.serialization[segment_start..self.serialization.len()]
+            };
```

Comment:
Why are we pushing the slash earlier now if we need to ignore it later?

Reply:
There's probably a much wiser way to do it, which I still need to figure out, but I'm making sure I push the root slash if there's one. I must admit it creates much more convoluted code afterwards. I ll try to refactor it.